unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarModifiers/AvatarModifierArea.cs

```diff
@@ -103,12 +104,20 @@ private void Update()
         foreach (GameObject avatarThatEntered in newAvatarsInArea.Except(avatarsInArea))
         {
             OnAvatarEnter?.Invoke(avatarThatEntered);
+            if (avatarThatEntered.tag.Equals("Player"))
```

Comment:
We dont use tags in the project, this approach is not bad per-se but it's something we usually do differently.

One of the problems with this implementation is that it leads to violation of the "interface segregation principle". You are using a general purpose system (like this area) to do stuff for an specific case (when your avatar is inside one of the areas).

I would move this code to `PlayerAvatarController`, at the moment you only have hooks there for the `HideArea` and not the `DisablePassport` but if you think about it, giving feedback about "passport disabled area" is pointless.

Disabled passports only affect you when clicking other player, if you are inside a `DisablePassport` area but a nearby player is outside, you will be able to open that person's passport (rendering the warning useless). And the other way around, if you are outside a `DisablePassport` area but a nearby player is within one: you won't be able to open the passport and you won't have any warning.

My suggestion would be to re-evaluate the need of that warning and in any case move this code to PlayerAvatarController and add support for `DisablePassport` (refactoring a bit `DisablePassportModifier` so instead of `AvatarShape` uses a handler like the other modifier).